src/CommunityToolkit.Maui.Core/Primitives/LayoutState.shared.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace CommunityToolkit.Maui.Core;
+
+/// <summary>
+/// Enum that represents possible layout states
+/// </summary>
+public enum LayoutState
```

Comment:
Do we need this class at all? can we assume the user always uses Custom state?

Reply:
It's probably not strictly necessary, just convenience. I do like having the preset enum built into the toolkit for common uses like Loading or Error, with Custom available for more advanced scenarios.

Reply:
Going with a string StateKey (always custom) for now, we can see how it feels.